html/admin/api/server/migrateAsset.php

```diff
@@ -0,0 +1,101 @@
+        //Create new manufacturer if needed
+        if ($manufacturer['instances_id'] != $_POST['instances_id'] && $manufacturer['instances_id'] != NULL) {
+            //See if they have a similarly named manufacturer
+            $DBLIB->where("manufacturers_name", "%" . $manufacturer['manufacturers_name'] . "%", 'LIKE');
+            $DBLIB->where("instances_id", $_POST['instances_id']);
+            $possibleManufacturer = $DBLIB->getOne("manufacturers", ["manufacturers.*"]);
+            if ($possibleManufacturer) {
+                $manufacturer = $possibleManufacturer;
+            } else {
+                $manufacturer['instances_id'] = $_POST['instances_id'];
+                $manufacturer['manufacturers_internalAdamRMSNote'] = "Created during asset migration";
+                unset($manufacturer['manufacturers_id']);
+                $manufacturer['manufacturers_id'] = $DBLIB->insert("manufacturers", $manufacturer);
+            }
+        } 
+
+        //Get Category
+        $DBLIB->where("assetCategories_id", $assetType['assetCategories_id']);
+        $DBLIB->where("assetCategories_deleted", 0);
+        $category = $DBLIB->getOne("assetCategories", ["assetCategories.*"]);
+        if (!$category) finish(false, ["code"=>"CATEGORY_NOT_FOUND", "message"=>"Category not found"]);
+        //Create new category if needed
+        if ($category['instances_id'] != $_POST['instances_id'] && $category['instances_id'] != NULL) {
+            //See if they have a similarly named category
+            $DBLIB->where("assetCategories_name", "%" . $category['assetCategories_name'] . "%", 'LIKE');
+            $DBLIB->where("instances_id", $_POST['instances_id']);
+            $possibleCategory = $DBLIB->getOne("assetCategories", ["assetCategories.*"]);
+            if ($possibleCategory) {
+                $category = $possibleCategory;
+            } else {
+                $category['instances_id'] = $_POST['instances_id'];
+                unset($category['assetCategories_id']);
+                $category['assetCategories_id'] = $DBLIB->insert("assetCategories", $category);
+            }
+        }
+
+        //Create AssetType
+        $assetType['instances_id'] = $_POST['instances_id'];
+        $assetType['manufacturers_id'] = $manufacturer['manufacturers_id'];
+        $assetType['assetCategories_id'] = $category['assetCategories_id'];
+        unset($assetType['assetTypes_id']);
+        $assetType['assetTypes_id'] = $DBLIB->insert("assetTypes", $assetType);
+    }
+}
+
+//Create Asset
+$asset['instances_id'] = $_POST['instances_id'];
+$asset['assetTypes_id'] = $assetType['assetTypes_id'];
+$asset['assets_storageLocation'] = NULL;
+$asset['assets_linkedTo'] = NULL; //We don't migrate linked assets
+$asset['assets_tag'] = generateNewTag(); //This is now a new asset so needs a unique tag
```

Comment:
I'd suggest people would want to keep the tags - especially if its printed on the asset